k2/torch/csrc/torch_utils.h

```diff
@@ -18,19 +19,44 @@
  * limitations under the License.
  */
 
-#ifndef K2_PYTHON_CSRC_TORCH_TORCH_UTIL_H_
-#define K2_PYTHON_CSRC_TORCH_TORCH_UTIL_H_
+#ifndef K2_TORCH_CSRC_TORCH_UTILS_H_
+#define K2_TORCH_CSRC_TORCH_UTILS_H_
 
+#include <memory>
 #include <string>
 
 #include "k2/csrc/array.h"
 #include "k2/csrc/device_guard.h"
 #include "k2/csrc/fsa.h"
 #include "k2/csrc/log.h"
 #include "k2/csrc/pytorch_context.h"
+#include "k2/torch/csrc/fsa_class.h"
+#include "k2/torch/csrc/ragged_any.h"
+#include "torch/torch.h"
 
 namespace k2 {
 
+struct FsaClass;
+
+// We need this wrapper so that we can convert an instance
+// of RaggedAny into `torch::IValue`
+struct RaggedAnyHolder : public torch::CustomClassHolder {
+  std::shared_ptr<RaggedAny> ragged = nullptr;  // not owned by this class
```

Comment:
```suggestion
  std::shared_ptr<RaggedAny> ragged;
```

Reply:
Can we use `RaggedAny ragged;` directly?

`RaggedAny` contains a set of shared pointers. Using a `std::shared_ptr` here
increases one level of indirection.